app/grandchallenge/archives/tasks.py

```diff
@@ -35,6 +38,34 @@ def add_images_to_archive(*, upload_session_pk, archive_pk, interface_pk=None):
         item.values.set([civ])
 
 
+@shared_task
+def add_images_to_archive_item(
+    *, upload_session_pk, archive_item_pk, interface_pk=None
+):
+    archive_item = ArchiveItem.objects.get(pk=archive_item_pk)
+    interface = ComponentInterface.objects.get(pk=interface_pk)
+
+    civ_pks_to_remove = set()
+    civ_pks_to_add = set()
+
+    civ = archive_item.values.filter(interface=interface).first()
+    if civ:
+        civ_pks_to_remove.add(civ.pk)
+
+    civ = ComponentInterfaceValue.objects.create(interface=interface)
+    civ_pks_to_add.add(civ)
+
+    add_images_to_component_interface_value(
+        component_interface_value_pk=civ.pk,
+        upload_session_pk=upload_session_pk,
+    )
+    update_archive_item_values(
+        archive_item_pk=archive_item_pk,
+        civ_pks_to_remove=list(civ_pks_to_remove),
+        civ_pks_to_add=list(civ_pks_to_add),
+    )
```

Comment:
It is better not to use the celery tasks directly, but instead to call them with `on_commit(<task>.apply_async(kwags={...}))`.

Reply:
Ok, understood, I went with a hybrid solution, calling `add_images_to_component_interface_value` asynchronously, but doing the achive item updating synchronously in the function. The archive item updating cannot be done asynchronously, because then calling the task a second time would (or might) result in two civs being created.

Reply:
Not sure that I quite follow - you're now using atomic transactions so there will be no changes until the entire task completes. If the method is called again then the existing interface will be removed, right?

Reply:
Or maybe we need to check at the start of `add_images_to_archive_item` that the image from the upload session does not already belong to that archive item with that interface?

Reply:
The way it is now it works. The second time the task is called the existing interface will be removed. When I tried to call `update_archive_item_values` aysnchronously, there was no new interface yet the second time the task ran, and so there was no interface to remove. Somehow the updating was running in parallel to the second run of the task...